src/hooks/permission.ts

```diff
@@ -0,0 +1,137 @@
+import { useQuery } from '@apollo/react-hooks'
+import gql from 'graphql-tag'
+import { groupBy } from 'ramda'
+import { useApp } from '../contexts/AppContext'
+import hasura from '../hasura'
+import { PermissionGroupProps } from '../types/general'
+
+export const useDefaultPermissions = () => {
+  const { loading, error, data, refetch } = useQuery<hasura.GET_ROLE_PERMISSION>(gql`
+    query GET_ROLE_PERMISSION {
+      role_permission {
+        id
+        role_id
+        permission_id
+      }
+    }
+  `)
+
+  const defaultRolePermissions =
+    data?.role_permission.reduce<
+      {
+        [roleId in string]?: string[]
+      }
+    >(
+      (accumulator, currentValue) => ({
+        ...accumulator,
+        [currentValue.role_id]: [...(accumulator[currentValue.role_id] || []), currentValue.permission_id],
+      }),
+      {},
+    ) || {}
+
+  return {
+    loadingPermissions: loading,
+    errorPermissions: error,
+    defaultRolePermissions,
+    refetchPermissions: refetch,
+  }
+}
+
+export const usePermissionGroupsCollection = () => {
```

Comment:
Group*s*Collection means each item of the collection are group*s*
```js
const groups = []
const groupsCollection = [groups, groups, groups...]

```

Reply:
```js
usePermissionGroups = () => {
}
usePermissionGroup = () => {
}

const App = () => {
   const group = usePermissionGroup()
   const groups = usePermissionGroups()
}
```